framework/web/CHttpRequest.php

```diff
@@ -712,8 +712,15 @@ public function getUrlReferrer()
 	 * Returns the user agent, null if not present.
 	 * @return string user agent, null if not present
 	 */
-	public function getUserAgent()
-	{
+	static function getUserHostAddress($proxy_set_header ='HTTP_X_FORWARDED_FOR')
+	{
```

Comment:
This is definetely incorrect change: why `getUserAgent()` removed and `getUserHostAddress()` is static?